domain/src/test/java/org/oppia/domain/topic/TopicControllerTest.kt

```diff
@@ -767,13 +793,344 @@ class TopicControllerTest {
     assertThat(questionListResultCaptor.value.isFailure()).isTrue()
   }
 
+  @Test
```

Comment:
Other tests I think should be added:

1. Verifying that finishing a single story sets both chapters to completed
2. Each true failure case for the new data providers (empty list probably shouldn't fail--see my comments below).

Reply:
For (1.) added new test case - `testCompletedStoryList_finishEntireStory_checkChapters_allAreCompleted`

For (2.) updated following test case-
`testCompletedStoryList_recordOneChapterProgress_completedStoryListIsEmpty`
`testCompletedStoryList_validData_withoutAnyProgress_completedStoryListIsEmpty`
`testOngoingTopicList_finishEntireTopic_ongoingTopicListIsEmpty`
`testOngoingTopicList_validData_withoutAnyProgress_ongoingTopicListIsEmpty`